### ISSUE TITLE
reproduce

### ISSUE THREAD

**Issue by hzphzp** [wgchang/DSBN#8]
Hi~ Thank you for your excellent work. 
I just reproduced the experiment for VisDA dataset with MSTN by running the script in Readme. But I got 76.14% mean Acc and 91.91% 84.46% 75.78% 60.04% 94.91% 30.75% 89.29% 81.58% 90.06% 78.47% 84.28% 52.20% Class Acc which are relatively lower than that reported in your paper but similar with the result reported for CPUA. I run the code in a V100 without changing any hyper-parameter. I am wondering if there is something I missed in stage2 training.  An early response would be appreciated.

**Comment by woozch** [comment]
Hi. 
I don't remember the details because it was a quite previous study, but I do remember running my experiment with the hyper parameter setting in the Readme. 

Since the domain adversarial loss is relatively unstable in unsupervised domain adaptation learning, if the result of stage 1 is not good, stage 2 is a refinement step, so I remember that there was a big difference in performance. This could be the downside of two-stage training.

The average performance of the learning models in stage 1 is relatively similar, but among them, you should use a model with overall good performance in the class to proceed with stage 2 to get good results.

**Comment by hzphzp** [comment]
Thank you for your reply. I found that the results reported in stage 1 for CPUA and MSTN are similar but the results in stage2 of MSTN are much higher than CPUA. However, the stage2 training for MSTN and CPUA are the same (only with classification loss of source and pseudo target domain, am I correct?). So, I am wondering what bring the difference between the stage 2 results of MSTN and CPUA? Any suggestions?

**Comment by woozch** [comment]
Hi. Sorry for the late reply.
Our algorithm aims to show performance improvement when the existing algorithm divides batch normalization by domain.
The performance marked in "reproduced" is the average accuracy without DSBN, and the performance marked in stage1 is the result when DSBN is used.
As mentioned in previous answers, we can see that for MSTN, the overall class performance is good, whereas for CPUA, the performance of a specific class is low. 
In the case of MSTN, the overall performance is uniformly high (above 40%~), so I think that the performance improvement in the refinement step was greater. While in the case of CPUA, it seems that there was the performance degradation of a specific class such as knife in the refinement step.

**Comment by hzphzp** [comment]
> Hi. Sorry for the late reply. Our algorithm aims to show performance improvement when the existing algorithm divides batch normalization by domain. The performance marked in "reproduced" is the average accuracy without DSBN, and the performance marked in stage1 is the result when DSBN is used. As mentioned in previous answers, we can see that for MSTN, the overall class performance is good, whereas for CPUA, the performance of a specific class is low. In the case of MSTN, the overall performance is uniformly high (above 40%~), so I think that the performance improvement in the refinement step was greater. While in the case of CPUA, it seems that there was the performance degradation of a specific class such as knife in the refinement step.

Thank you~ Got it.
